### PR TITLE
Use poetry for packaging

### PR DESCRIPTION
**Pull request by carlio**
This converts prospector into using poetry for packaging instead of setuptools

I created to releases to test here:
* https://github.com/PyCQA/prospector/releases/tag/1.5.0-dev1
* https://github.com/PyCQA/prospector/releases/tag/1.5.0-dev2

Note that I had to drop testing with python `3.10-dev` because it cannot install `pep8` due to a JSON decode error. I also dropped `safety` because it is not compatible with the output of `pip freeze` with the way that poetry sets versions.

(pre-cursor towards #376 )

### CODE REVIEW

**Comment by carlio** [comment]
@Pierre-Sassoulas FYI

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -20,6 +20,7 @@ repos:
       - id: codespell
         args:
             - --ignore-words-list=nin
+            - --skip=poetry.lock
```
Comment (carlio) [inline .pre-commit-config.yaml:23]: Some dependencies have spelling which codespell complains about, nothing we can do about that!

File: pyproject.toml [hunk 2]
```diff
@@ -1,3 +1,84 @@
+[tool.poetry]
+name = "prospector"
+version = "1.5.0.dev2"
+description = ""
+authors = ["Carl Crowder <git@carlcrowder.com>"]
+maintainers = ["Carl Crowder <git@carlcrowder.com>",
+               "Carlos Coelho <carlospecter@gmail.com>"]
```
Comment (carlio) [inline pyproject.toml:7]: @Pierre-Sassoulas please add yourself to this list if you want to! Also to `CONTRIBUTERS.rst`.

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -163,7 +159,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: [3.6, 3.7, 3.8, 3.9, 3.10-dev]
```
Comment (carlio) [inline .github/workflows/ci.yaml:166]: 3.10-dev simply can't install all dependencies when using poetry, so let's wait until 3.10 is released properly before attempting to fix that, or if people are using the 3.10-rc maybe they will create reproduction steps and raise issues.

File: pyproject.toml (hunk 2 above)
Comment (Pierre-Sassoulas) [reply pyproject.toml:7]:
```suggestion
               "Carlos Coelho <carlospecter@gmail.com>",
               "Pierre Sassoulas <pierre.sassoulas@gmail.com>"]
```

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (Pierre-Sassoulas) [reply .github/workflows/ci.yaml:166]: I added 3.10 in the meta-information in the ``setup.py`` it's probably best to remove it then.

File: pyproject.toml (hunk 2 above)
Comment (carlio) [reply pyproject.toml]: :+1:

**Pull request merged**